prompt2model/utils/config.py

```diff
@@ -0,0 +1,14 @@
+"""Place to store all the default configurations."""
+MAX_SUPPORTED_BATCH_SIZE = 4
+
+DEFAULT_HYPERPARAMETERS_SPACE = {
+    "min_num_train_epochs": 10,
+    "max_num_train_epochs": 20,
+    "save_strategy": ["epoch", "steps", "no"],
+    "evaluation_strategy": ["epoch", "no"],
```

Comment:
Does it make sense to search over these two hyperparameters? Intuitively it seems like these wouldn't affect the model's training (in which case we can keep them as a constant value, e.g. `epoch`)

Reply:
Also, it seems that `evaluation_strategy` is not used in the Optuna parameter tuning, so I think we should definitely be setting that one as a constant. (I also think we should set `save_strategy` as a constant, fyi)

Reply:
Yes, you are right that the current set of strategies would not provide any further optimizations in terms of qualitative search.

However, I feel like I would take another pass to see if some other strategies exist. Because by theory, the save of weights after each epoch would be less effective compared to the saving of checkpoint where the evaluation loss is the least. So, if those kind of strategies exist, then I would try to add those, else will keep this to constant i.e. `epoch`.

Reply:
> because by theory, the save of weights after each epoch would be less effective compared to the saving of checkpoint where the evaluation loss is the least

That makes sense, but do these save strategies actually affect whether we choose the final checkpoint as the one where the evaluation loss is the least?
 
`epoch` just saves the model after a full epoch of training, while `steps` saves a model after a specified number of steps. As I understand it, the latter just saves many more intermediate checkpoints, while the former saves fewer checkpoints.

The final model returned will still be the one at the end of the full training process.

As such, I think it makes sense to use the value `no` here.

Reply:
Yeah that does makes a good point. Making those changes.